integration/setup/setup.go

```diff
@@ -133,13 +137,31 @@ func SetupWithOpts(tb testing.TB, opts *SetupOpts) *SetupResult {
 	logger := testutil.LevelLogger(tb, level)
 
 	var client *mongo.Client
-	var uri string
+	uri := *targetURLF
 
 	if *targetURLF == "" {
 		client, uri = setupListener(tb, setupCtx, logger)
-	} else {
-		client = setupClient(tb, setupCtx, *targetURLF)
-		uri = *targetURLF
+	}
+
+	if opts.ExtraOptions != nil || *targetURLF != "" {
+		u, err := url.Parse(uri)
+		require.NoError(tb, err)
+
+		q := u.Query()
+
+		for k, vs := range opts.ExtraOptions {
+			for _, v := range vs {
+				q.Set(k, v)
+			}
+		}
+
+		u.RawQuery = q.Encode()
+		uri = u.String()
+
+		// If ExtraOptions is set for in-process FerretDB, two clients are created
+		// on purpose. setupListener creates a client to check in-process FerretDB
```

Comment:
No, there is no purpose now :)

Reply:
@chilagrow PTAL at my changes

Reply:
Thank you for refactoring that part of the code :+1: